internal/controllers/gateway/route_utils.go

```diff
@@ -897,3 +897,51 @@ func isParentRefEqualToParent[
 
 	return true
 }
+
+func isRouteAcceptedByListener[T types.RouteT](ctx context.Context,
```

Comment:
A doc comment would be helpful here as well. The name seems self explanatory but I'd be scratching my head wondering why do I need to provide the gateway and parentRef additionally to the listener.